go/interpreter/lfvm/interpreter_test.go

```diff
@@ -626,28 +624,57 @@ func TestSteps_StaticContextViolation(t *testing.T) {
 	}
 }
 
-// FIXME: rewrite as static gas check (for all opcodes)
-func TestStepsFailsOnTooLittleGas(t *testing.T) {
-	ctxt := getEmptyContext()
-	instructions := []Instruction{
-		{PUSH1, 0},
-	}
+func TestSteps_FailsWithLessGasThanStaticCost(t *testing.T) {
 
-	ctxt.params = tosca.Parameters{
-		Input:  []byte{},
-		Static: false,
-		Gas:    2,
-		Code:   []byte{0x0},
+	// Ignored because they have static cost 0
+	ignore := []OpCode{
+		STOP,
+		RETURN,
+		REVERT,
+		INVALID,
+		JUMP_TO,
+		SSTORE,
 	}
-	ctxt.gas = 2
-	ctxt.code = instructions
 
-	_, err := steps(&ctxt, false)
-	if want, got := errOutOfGas, err; want != got {
-		t.Errorf("unexpected error: want %v, got %v", want, got)
+	for _, op := range allOpCodesWhere(isValidOpCode) {
+		if slices.Contains(ignore, op) {
+			continue
+		}
+
+		t.Run(op.String(), func(t *testing.T) {
+			ctxt := getEmptyContext()
+			staticGasPrices := getStaticGasPrices(tosca.R07_Istanbul)
+
+			ctxt.params = tosca.Parameters{
+				Input:  []byte{},
+				Static: false,
+				Code:   []byte{0x0},
+			}
+			ctxt.code = []Instruction{{op, 0}}
+			ctxt.stack.stackPointer = 50
+			ctxt.gas = staticGasPrices.get(op) - 1
+
+			if ctxt.gas < 0 {
+				t.Fatalf("invalid test setup: gas is negative")
+			}
```

Comment:
Maybe, instead of filtering out explicit instructions with zero-cost here, just check the `staticGasPrices` for zero here and return early if so. Then you are automatically sure that you got the right instructions (unless wanting to verify this is part of the test's intention)

Reply:
In a later change, we figured out that it was important to check gas for each revision and zero was then more relevant, as it is the static cost of some "valid" ops after Berlin. 
The new code skips OP/Revision test when the expected static gas cost is zero.